service/payment/src/main/java/com/ttbmp/cinehub/service/payment/stripe/StripeServicePaymentFacade.java

```diff
@@ -0,0 +1,168 @@
+package com.ttbmp.cinehub.service.payment.stripe;
+
+import com.stripe.exception.StripeException;
+import com.stripe.model.Customer;
+import com.stripe.model.PaymentIntent;
+import com.stripe.model.PaymentIntentCollection;
+import com.stripe.model.PaymentMethod;
+import com.ttbmp.cinehub.core.dto.TicketDto;
+import com.ttbmp.cinehub.core.dto.UserDto;
+import com.ttbmp.cinehub.core.entity.Payment;
+import com.ttbmp.cinehub.core.service.payment.PaymentService;
+import com.ttbmp.cinehub.core.service.payment.request.PayServiceRequest;
+
+import java.util.List;
+
+/**
+ * @author Palmieri Ivan
+ */
+public class StripeServicePaymentFacade implements PaymentService {
+
+    private final ServiceRequestStripe serviceRequestStripe;
+
+    String errorStripe;
+
+    public StripeServicePaymentFacade() {
+        this.serviceRequestStripe = new ServiceRequestStripe();
+    }
+
+    @Override
+    public boolean pay(PayServiceRequest payServiceRequest) {
+        Customer customer = null;
+        try {
+            customer = serviceRequestStripe.isExistent(payServiceRequest.getUserDto());
+        } catch (StripeException e) {
+            e.printStackTrace();
+        }
+        if (customer == null) {
+            errorStripe = "Error while retrieving the customer";
+            return false;
+        }
+        PaymentMethod paymentMethod = null;
+        try {
+            paymentMethod = serviceRequestStripe.retrieveCard(customer);
+        } catch (StripeException e) {
+            e.printStackTrace();
+        }
+        if (paymentMethod == null) {
+            errorStripe = "Error while retrieving paper";
+            return false;
+        }
+        try {
+            if (serviceRequestStripe.updateBalance(customer, payServiceRequest.getTicketPrice())) {
+                if (!serviceRequestStripe.setBalance(customer, payServiceRequest.getTicketPrice(), paymentMethod)) {
+                    errorStripe = "Problem updating your account";
+                }
+            } else {
+                errorStripe = "Error during payment, or insufficient money";
+                return false;
+            }
+        } catch (StripeException e) {
+            e.printStackTrace();
+        }
+
+        return true;
+
+    }
+
+    /*Refund a specific transaction using the id*/
+    @Override
+    public boolean refoundById(UserDto userDto, String id) {
```

Comment:
Mixing the business logic with the error handling logic decrease the readability of the code.